stock_picking_invoice_product_group/wizards/stock_invoice_onshipping.py

```diff
@@ -0,0 +1,58 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016 Alex Comba - Agile Business Group
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+
+from openerp import models, api, fields
+
+
+class StockInvoiceOnshipping(models.TransientModel):
+    _inherit = 'stock.invoice.onshipping'
+
+    group_type = fields.Selection(
+        [('group_by_product', 'Group by product'),
+         ('group_by_product_category', 'Group by product category')
+         ],
+        string='Group Type',
+    )
+
+    @api.model
+    def _group_moves_by_group_type(self, moves, group_type):
+        """Return moves grouped by group_type"""
+        grouped_moves = {}
+        for move in moves:
+            partner = move.picking_id.partner_id
+            product = move.product_id
+            categ = product.categ_id
+            if group_type == 'group_by_product':
+                key = (partner.id, product.id)
+            else:
```

Comment:
I'd prefer to leave the possibility to choose 'Group by partner' option, this is the reason for that kind of implementation.